File: MyProjects/MAP/Assignment 7/Interpreter/Model/Expressions/ArithmExpr.cs
using System;
using System.Collections.Generic;
using Interpreter.Exceptions;
using System.Text;

namespace Interpreter.Model.Expressions
{
    public class ArithmExpr:IExpr
    {
        private IExpr left, right;
        private string oper;

        public ArithmExpr(string op, IExpr l, IExpr r)
        {
        oper = op;
            left = l;
            right = r;
        }

        public int evaluate(Utilities.IDictionary<string, int> dic)
        {
            int left = this.left.evaluate(dic);
            int right = this.right.evaluate(dic);

            switch (oper){
            case "+":
                return left + right;
            case "-":
                return left - right;
            case "*":
                return left * right;
            case "/":
                if (right == 0)
                    throw new DivisionByZero("Division by 0 is not allowed!");
                return left / right;
            }
            throw new InvalidOperator("Invalid operator!");
        }

        public override string ToString()
        {
            return "" + left + oper +right;
        }
    }
}

File: MyProjects/MAP/Practical Exam/Repeat Until Stmt/src/Model/Expressions/ArithmExpr.java
package Model.Expressions;

import Model.Utilities.IDictionary;
import Exception.*;

public class ArithmExpr implements IExpr {

    private IExpr left, right;
    private String operator;

    public ArithmExpr(String op, IExpr l, IExpr r){
        operator = op;
        left = l;
        right = r;
    }

    @Override
    public int evaluate(IDictionary<String, Integer> dic, IDictionary<Integer, Integer> heap) {
        int left = this.left.evaluate(dic,heap);
        int right = this.right.evaluate(dic,heap);

        switch (operator){
            case "+":
                return left + right;
            case "-":
                return left - right;
            case "*":
                return left * right;
            case "/":
                if(right == 0)
                    throw new DivisionByZero("Division by 0 is not admitted!");
                return left/right;
        }
        throw new InvalidOperator("Invalid operator!");
    }

    public String toString(){
        return "" + left + operator + right;
    }
}
